clients/bucket/list.go

```diff
@@ -29,16 +34,87 @@ func (c Client) List(ctx context.Context, params *BucketsListParams) error {
 	if params.OrgID == "" && params.OrgName == "" {
 		req = req.Org(c.ActiveConfig.Org)
 	}
-	if params.Name != "" {
-		req = req.Name(params.Name)
+	if params.Name != "" || params.ID != "" {
+		return c.findOneBucket(params, req)
+	}
+
+	printOpts := bucketPrintOptions{}
+	limit := params.PageSize
+	if params.Limit != 0 && params.Limit < limit {
+		limit = params.Limit
+	}
+	offset := params.Offset
+	after := params.After
+
+	// To guard against infinite loops from bugs in/incomplete implementations of server-side pagination,
+	// we track the IDs we see in each request and bail out if we get a duplicate.
+	seenIds := map[string]struct{}{}
+	if after != "" {
+		seenIds[after] = struct{}{}
+	}
+
+	for {
+		req := req
+		if limit != 0 {
+			req = req.Limit(int32(limit))
+		}
+		if offset != 0 {
+			req = req.Offset(int32(offset))
+		}
+		if after != "" {
+			req = req.After(after)
+		}
+
+		res, err := req.Execute()
+		if err != nil {
+			return fmt.Errorf("failed to list buckets: %w", err)
+		}
+		var buckets []api.Bucket
+		if res.Buckets != nil {
+			buckets = *res.Buckets
+		}
+		bailOut := false
+		for _, b := range buckets {
+			id := *b.Id
+			if _, alreadySeen := seenIds[id]; alreadySeen {
+				bailOut = true
```

Comment:
Ah yeah I see that hitting this condition would actually be expected if you were running the CLI with an older version that didn't have the pagination fully implemented, so it would not be good to error out on that!

Sounds like in order for the CLI to work with older versions that don't support the `after`-based pagination (like OSS `2.0.9`) the only option is to use `offset`-based pagination right now?

...Or maybe some kind of feature-flag approach where the server could indicate that it supports `after`-based pagination. That seems like a bit of a can of worms but probably something we're going to have to tackle eventually.